src/containers/oracle.js

```diff
@@ -66,124 +90,128 @@ class OraclePage extends React.Component {
+
+
+    // TODO: For any error case we will render an Oracle not found page
   }
 
   componentWillUnmount() {
-    console.log('componentWillUnmount');
-
-    this.props.onClearBetReturn();
+    this.props.onClearRequestReturn();
   }
 
   onRadioGroupChange(evt) {
-    console.log(`Radio value change ${evt.target.value}`);
-
     this.setState({
       radioValue: evt.target.value,
     });
   }
 
-  /** Confirm button on click handler passed down to CardVoting */
-  onBet(obj) {
+  /** The right card Confirm Button event handler * */
+  onConfirmBtnClicked(obj) {
     const { oracle, radioValue } = this.state;
-
-    const { walletAddrs, walletAddrsIndex } = this.props;
-
+    const {
+      onBet, onSetResult, onVote, onFinalizeResult, onApprove,
+    } = this.props;
+    const senderAddress = this.getCurrentSenderAddress();
     const selectedIndex = oracle.optionIdxs[radioValue - 1];
     const { amount } = obj;
 
-    const senderAddress = walletAddrs[walletAddrsIndex].address;
-
-    const contractAddress = 'fe99572f3f4fbd3ad266f2578726b24bd0583396';
-    console.log(`contractAddress is ${contractAddress}, selectedIndex is ${selectedIndex}, amount is ${amount}, senderAddress is ${senderAddress}`);
-
-    this.props.onBet(contractAddress, selectedIndex, amount, senderAddress);
-  }
-
-  onSetResult() {
-    const { oracle, radioValue } = this.state;
-    const { walletAddrs, walletAddrsIndex } = this.props;
-
+    // contractAddress should be the address of this oracle
+    // TODO: Update this to oracle.address for testing on testnet
     const contractAddress = '9697b1f2701ca9434132723ee790d1cb0ab0e414';
-    const selectedIndex = oracle.optionIdxs[radioValue - 1];
-    const senderAddress = 'qKjn4fStBaAtwGiwueJf9qFxgpbAvf1xAy';
-    console.log(`contractAddress is ${contractAddress}, selectedIndex is ${selectedIndex}, senderAddress is ${senderAddress}`);
 
-    this.props.onSetResult(contractAddress, selectedIndex, senderAddress);
+    if (this.state.config.name === 'BETTING') {
+      onBet(contractAddress, selectedIndex, amount, senderAddress);
+    } else if (this.state.config.name === 'SETTING') {
+      /** Result setter needs to have 100 BOT and get approved by Bodhi_token contract to use them* */
+      onApprove(oracle.topicAddress, ORACLE_BOT_THRESHOLD, senderAddress);
+
+      setTimeout(() => {
+        onSetResult(contractAddress, selectedIndex, senderAddress);
+      }, SUB_REQ_DELAY);
+    } else if (this.state.config.name === 'VOTING') {
+      /** The amount of voting needs to be approved by Bodhi_token * */
+      onApprove(oracle.topicAddress, amount, senderAddress);
+
+      setTimeout(() => {
+        onVote(contractAddress, selectedIndex, amount, senderAddress);
+      }, SUB_REQ_DELAY);
+
+      // The last Vote that hits BOT threshold needs to issue a finalize() after certain delay
+      if (oracle.amounts[selectedIndex] + amount >= ORACLE_BOT_THRESHOLD) {
```

Comment:
I am confused by this logic. Why is it calling `finalizeResult()` if it passes the threshold? Is this for the demo?

Normally, only finalizeResult() is called after the DecentralizedOracle does not hit the threshold within the `arbitrationEndBlock` period. This sets the final result in `TopicEvent`.

Reply:
Updated with new logic which checks if current block is greater than arbitrationEndBlock